### PR TITLE
feat: add `tx` and `txs` commands

### PR DESCRIPTION
**Pull request by dmvict**

### CODE REVIEW

File: gaia-rs/src/client.rs [hunk 1]
```diff
@@ -58,6 +61,10 @@ pub enum GaiaQueryCommands {
     Staking(StakingQueryCli),
     /// Querying commands for the ibc module
     Ibc(IbcQueryCli),
+    /// Querying tendermint for a transaction
+    Tx(TxQueryCli),
+    /// Querying tendermint for transactions by event
+    Txs(TxsQueryCli),
```
Comment (joneskm) [inline gaia-rs/src/client.rs:67]: Would it be possible to have the new commands added automatically inside gears?

File: x/auth/src/client/cli/tx_query.rs [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+            _ => Err("Unknown transaction type".to_string()),
+        }
+    }
+}
+
+impl Display for TxQueryType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TxQueryType::Hash => write!(f, "hash"),
+            TxQueryType::AccSeq => write!(f, "acc_seq"),
```
Comment (NYBACHOK) [inline x/auth/src/client/cli/tx_query.rs:71]: Check `strum` crate

File: x/auth/src/client/cli/tx_query.rs [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+        &self,
+        command: &Self::QueryCommands,
+    ) -> anyhow::Result<Self::QueryRequest> {
+        let TxsQueryCli {
+            events,
+            page,
+            limit,
+        } = command;
+
+        let events_str = events.trim_start_matches('\'').trim_end_matches('\'');
```
Comment (NYBACHOK) [inline x/auth/src/client/cli/tx_query.rs:271]: newtype for it?

File: x/auth/src/client/cli/tx_query.rs [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+        &self,
+        query: Self::QueryRequest,
+        node: url::Url,
+        _height: Option<gears::tendermint::types::proto::block::Height>,
+    ) -> anyhow::Result<Vec<u8>> {
+        let client = HttpClient::new(node.as_str())?;
+        let runtime = tokio::runtime::Runtime::new().expect("failed to create tokio runtime");
+
+        match query {
+            Self::QueryRequest::Tx(_) => unreachable!(),
```
Comment (NYBACHOK) [inline x/auth/src/client/cli/tx_query.rs:304]: unreachable?

File: x/auth/src/client/query.rs [hunk 5]
```diff
@@ -99,6 +97,104 @@ impl From<QueryAccountsResponse> for inner::QueryAccountsResponse {
     }
 }
 
+/// QueryParamsResponse is the response type for the Query/Params RPC method
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize, Query)]
```
Comment (NYBACHOK) [inline x/auth/src/client/query.rs:101]: I hope that my PR would be merged first

File: x/auth/src/client/cli/tx_query.rs (hunk 4 above)
Comment (dmvict) [reply x/auth/src/client/cli/tx_query.rs]: I'm reusing enum for requests and this variant is not used in the command.

File: gaia-rs/src/lib.rs [hunk 6]
```diff
@@ -60,9 +64,21 @@ impl ApplicationInfo for GaiaApplication {
 
 pub struct GaiaCore;
 
-pub struct GaiaCoreClient;
+pub struct GaiaCoreClient<M: TxMessage> {
+    tx_query_handler: TxQueryHandler<M>,
```
[the remark this answers is not in the document]
Comment (dmvict) [reply gaia-rs/src/lib.rs]: It's for deserializing concrete transaction types. It's integrated into gears, so we don't have this change.

File: x/auth/src/client/cli/tx_query.rs (hunk 4 above)
Comment (dmvict) [reply x/auth/src/client/cli/tx_query.rs]: fixed

File: x/auth/src/client/cli/tx_query.rs (hunk 3 above)
Comment (dmvict) [reply x/auth/src/client/cli/tx_query.rs]: added

File: x/auth/src/client/cli/tx_query.rs (hunk 2 above)
Comment (dmvict) [reply x/auth/src/client/cli/tx_query.rs]: used

File: gaia-rs/src/client.rs (hunk 1 above)
Comment (dmvict) [reply gaia-rs/src/client.rs]: moved commands to gears

File: gears/src/cli/query.rs [hunk 7]
```diff
@@ -14,10 +16,58 @@ pub struct CliQueryCommand<C: Subcommand> {
...
+    #[strum(serialize = "hash")]
+    Hash,
+    #[strum(serialize = "acc_seq")]
+    AccSeq,
+    #[strum(serialize = "signature")]
+    Signature,
+}
+
+impl std::fmt::Display for TxQueryType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
```
Comment (NYBACHOK) [inline gears/src/cli/query.rs:61]: You may remove this

File: gears/src/cli/query.rs [hunk 8]
```diff
@@ -14,10 +16,58 @@ pub struct CliQueryCommand<C: Subcommand> {
...
+pub struct TxsQueryCli {
+    #[arg(long)]
+    pub events: String,
+    #[arg(long, default_value_t = 1)]
+    pub page: u32,
+    #[arg(long, default_value_t = 30)]
+    pub limit: u32,
+}
+
+#[derive(Debug, Clone, strum::EnumString)]
```
Comment (NYBACHOK) [inline gears/src/cli/query.rs:50]: Display is more appropriate

File: gears/src/cli/query.rs (hunk 7 above)
Comment (dmvict) [reply gears/src/cli/query.rs]: removed

File: gears/src/cli/query.rs (hunk 8 above)
Comment (dmvict) [reply gears/src/cli/query.rs]: added

**Pull request merged**